tasks/sendwithus.coffee

```diff
@@ -59,85 +62,54 @@ class Sendwithus
 
   # API handler
   api: (path, options = {}) ->
-    url = @buildURL(path)
-    # grunt.log.writeln 'Request url:', url
     defaultOptions =
       method: 'GET'
-      url: url
+      url: @buildURL(path)
       auth:
         user: @apiKey
         password: ''
       headers:
         accept: 'application/json'
 
     requestOptions = _.merge {}, defaultOptions, options
+    requestPromise = requestp(requestOptions)
 
-
-    return requestp(requestOptions).then(
-      (response) ->
-        return response
-
+    return requestPromise.then(
+      (res) ->
+        return res
       (err) ->
-        grunt.log.error 'API request failed with statusCode:', err.statusCode
-        return err
```

Comment:
This, as it turns out, was a big source of the problem. Because I was pushing the error back into the promise chain, I would have to catch it in every single promise termination in order to stem the errors. I moved this up into the API call and throw an error there so that it stops error propagation unless you manually invoke `catch` in the promise chain.